service/models/operations/optimize.py

```diff
@@ -59,13 +59,26 @@ def gen_risk_bound(self):
             return -self.risk_bound
 
 
-def objfun(x, initial_guess, objective_function_option):
-    if objective_function_option == "lower_bound":
-        return np.sum(np.abs(x))
-    if objective_function_option == "upper_bound":
-        return -np.sum(np.abs(x))
-    if objective_function_option == "initial_guess":
-        return np.sum(np.abs(x - initial_guess))
+def objfun(x, initial_guess, objective_function_option, relative_importance):
+    total_sum = 0
+    sum_of_all_weights = np.sum(relative_importance) or 1  # fallback to 1 if sum is 0
+    if (
+        x is None
+        or initial_guess is None
+        or objective_function_option is None
+        or relative_importance is None
+    ):
+        return total_sum
```

Comment:
Is this a fail case?
If it is, did we want to run our optimize under this fail case? 
If we do want to run the optimize we should include some logging for awareness to readers of this code and the logs
If we do not want to run the optimize due to this fail case we probably want to error out?

Reply:
thanks, we would want to error out in this case, have added